File: Adapter/PHP/LogAdapter.php
<?php
  class LogAdapter implements LogDbOperateAPI {
    private $logFileOperate;

    public function LogAdapter($logFileOperate) {
      $this->logFileOperate = $logFileOperate;
    }

    public function createLog($logger) {
      $content = $this->logFileOperate->readLogFile();
      $this->logFileOperate->writeLogFile("id: " . $logger->getLogId() . ";ope_user_id: " . $logger->getOpeUserId() . ";content: " . $content);
      echo "write into db\n";
    }
  }

File: Adapter/Go/adapter/LogAdapter.go
package adapter

import "fmt"

type LogAdapter struct {
  LogFileOperateAPI
  *LogFileOperate
}

func NewLogAdapter(logFileOperate *LogFileOperate) *LogAdapter {
  return &LogAdapter{LogFileOperate: logFileOperate}
}

func (this *LogAdapter) CreateLog(lg *Logger) {
  content := this.LogFileOperate.ReadLogFile()
  this.LogFileOperate.WriteLogFile("id: " + lg.GetLogId() + ";ope_user_id: " + lg.GetOpeUserId() + ";content: " + content);
  fmt.Printf("Write into db\n");
}
